feature/center/src/main/java/com/mifos/feature/center/center_list/ui/CenterListScreen.kt

```diff
@@ -0,0 +1,578 @@
+import com.mifos.core.designsystem.component.MifosSweetError
+import com.mifos.core.designsystem.theme.Black
+import com.mifos.core.designsystem.theme.BlueSecondary
+import com.mifos.core.designsystem.theme.DarkGray
+import com.mifos.core.designsystem.theme.LightGray
+import com.mifos.core.designsystem.theme.White
+import com.mifos.core.objects.group.Center
+import com.mifos.core.ui.components.SelectionModeTopAppBar
+import com.mifos.feature.center.R
+import kotlinx.coroutines.flow.flowOf
+
+@Composable
+fun CenterListScreen(
+    createNewCenter: () -> Unit,
+    syncClicked: (List<Center>) -> Unit,
+    onCenterSelect: (Center) -> Unit
+) {
+
+    val viewModel: CenterListViewModel = hiltViewModel()
+    val refreshState by viewModel.isRefreshing.collectAsStateWithLifecycle()
+    val state by viewModel.centerListUiState.collectAsStateWithLifecycle()
+
+    LaunchedEffect(key1 = true) {
+        viewModel.getCenterList()
+    }
+
+    CenterListScreen(
+        state = state,
+        createNewCenter = createNewCenter,
+        onRefresh = {
+            viewModel.refreshCenterList()
+        },
+        refreshState = refreshState,
+        onCenterSelect = onCenterSelect,
+        syncClicked = syncClicked
+    )
+}
+
+@Composable
+fun CenterListScreen(
+    state: CenterListUiState,
+    createNewCenter: () -> Unit,
+    onRefresh: () -> Unit,
+    refreshState: Boolean,
+    syncClicked: (List<Center>) -> Unit,
+    onCenterSelect: (Center) -> Unit
+) {
+
+    val snackbarHostState = remember { SnackbarHostState() }
+    var isInSelectionMode by rememberSaveable { mutableStateOf(false) }
+    val selectedItems = remember { mutableStateListOf<Center>() }
+    val resetSelectionMode = {
+        isInSelectionMode = false
+        selectedItems.clear()
+    }
+    BackHandler(enabled = isInSelectionMode) {
+        resetSelectionMode()
+    }
+
+    val pullRefreshState = rememberPullRefreshState(
+        refreshing = refreshState,
+        onRefresh = onRefresh
+    )
+
+    LaunchedEffect(
+        key1 = isInSelectionMode,
+        key2 = selectedItems.size,
+    ) {
+        if (isInSelectionMode && selectedItems.isEmpty()) {
+            isInSelectionMode = false
+        }
+    }
+
+    MifosScaffold(
+        modifier = Modifier,
+        topBar = {
+            if (isInSelectionMode) {
+                SelectionModeTopAppBar(
+                    itemCount = selectedItems.size,
+                    syncClicked = { syncClicked(selectedItems.toList()) },
+                    resetSelectionMode = resetSelectionMode
+                )
+            }
+        },
+        snackbarHostState = snackbarHostState,
+        bottomBar = { },
+        floatingActionButton = {
+            FloatingActionButton(
+                onClick = { createNewCenter() },
+                containerColor = BlueSecondary
+            ) {
+                Icon(
+                    imageVector = Icons.Filled.Add,
```

Comment:
Do we have common file for all the icons

Reply:
I used material3 icons, thus no need to add to the common file